src/main/java/uk/gov/companieshouse/service/dissolution/DissolutionCreator.java

```diff
@@ -29,10 +29,10 @@ public DissolutionCreator(
         this.responseMapper = responseMapper;
     }
 
-    public DissolutionCreateResponse create(DissolutionCreateRequest body, String companyNumber, String userId, String ip, String email) {
+    public DissolutionCreateResponse create(DissolutionCreateRequest body, String companyNumber, String companyName, String userId, String ip, String email) {
```

Comment:
You could pass through the company profile instead of passing company name and number through separately and let the mapper extract them.